src/main/java/jenkins/MasterToSlaveFileCallable.java

```diff
@@ -7,6 +7,7 @@
 
 /**
  * {@link FileCallable}s that are meant to be only used on the master.
+ * @deprecated Use a core dependency of 1.580.1 or later rather than this library.
  */
 public abstract class MasterToSlaveFileCallable<T> implements FileCallable<T>, RoleSensitive {
```

Comment:
According to Oracle documentation http://docs.oracle.com/javase/7/docs/technotes/guides/javadoc/deprecation/deprecation.html both are required but I did not mean to claim a bug, I just wanted to understand.
Anywho, I do not think it is of big impact, so :bee: